twisted/ldaptor
# Use absolute imports and fix all python2 -3 warnings when running tests.

Scope
====

This update the code so that when running the tests with python2 -3 we don't get any deprecation warnings.

It also updates the usage of absolute imports.

This is a work toward #55

Changes
=======

The deprecation warning for running reactor.iterate() inside the test is still there.

It adds six as a dependecy to make it easier to port.
For now ldaptor.compat is still used. It can be removed in a final cleanup but for now I want to focus on the important parts. 

I think that six was already present in all ldaptor env as it might be a dependency of some of the existent direct dependencies.

------

with `op` and `foo2` and `a` and `b` I am using `sut` (System Under Test` for the variable holding the target of the test. For me is easier to identify what is tested.

I have also refactored some tests to use local setup/arrangement as it make it easier to read.

No need to switch between 2 methods to see how the tests works.

With local setup is also easier to make sure that setUp does not have unused code.... TestOperations was a good example of setUp code not used in any test.

------

While working at writing missing tests  I found some bugs, so I wrote more tests to make sure we are ok.


How to test
=========

I don't know how to update tox to pass arguments to python interpretor.

To test this I was doing.

```
. .tox/py27-twtrunk/bin/activate
python2 -3 .tox/py27-twtrunk/bin/trial ldaptor
```

and then check for warnings in ldaptor code.

This should be merged only after parent is merged and then re-targeted to be merged in master.
